src/plugins/visualizations/public/embeddable/visualize_embeddable.tsx

```diff
@@ -224,6 +232,31 @@ export class VisualizeEmbeddable extends Embeddable<VisualizeInput, VisualizeOut
   // @ts-ignore
   hasInspector = () => Boolean(this.getInspectorAdapters());
 
+  onContainerLoading = () => {
+    this.domNode.setAttribute('data-render-complete', 'false');
+    this.domNode.setAttribute('data-loading', '');
+    this.domNode.removeAttribute('data-error');
+    ReactDOM.render(<EmbeddableErrorLabel />, this.labelNode);
+  };
+
+  onContainerRender = (count: number) => {
+    this.domNode.removeAttribute('data-loading');
+    this.domNode.setAttribute('data-render-complete', 'true');
+    this.domNode.setAttribute('data-rendering-count', count.toString());
+    dispatchRenderComplete(this.domNode);
+  };
+
+  onContainerError = (error: RenderError) => {
+    if (this.abortController) {
```

Comment:
This is related to: https://github.com/elastic/kibana/pull/68623/files#r437278595

I guess here could be a race condition, that when error pops up, `this.abortController` is actually an abortController for new on-going request which haven't finished yet?

Reply:
The graying out will actually prevent the user from applying a filter while another search is running, reducing the chances of this happening.